### PR TITLE
45 perception erster umsetzungsplan

### LINKED ISSUE #27: Minimal Viable Vehicle
### Description

Fahren
- **Mit "leeren" Grundkomponenten erstellen**
- Erstmal geradeaus fahren
- Throttle setzen
- Anfängliche Sensor Grundkonfiguration
- Auto aus Carla Leaderboard auswählen

### LINKED ISSUE #45: Perception: Erster Umsetzungsplan
Welche Bausteine aus der Recherche sollen verwendet werden?

Ziel: Markdown-Dokument mit folgendem Inhalt:
High-level, pro Aufgabe die von der Komponente gelöst werden soll:
- Welche konkrete Umsetzungsmöglichkeit wird gewählt
- grobe nächste Issues/Milestones zur Umsetzung

### PR DESCRIPTION
**Pull request by MarcoRiedenauer**
# Description

This PR summarizes the modules and possible implementations of the perception plus future issues/milestones.

Fixes #45 
## Time invested

15h @MarcoRiedenauer  

## Type of change

Documentation

## Does this PR introduce a breaking change?

No, just docu.

## Most important changes

- Possible implementations and algorithms for perception modules
- Future milestones/issues

# Checklist: (WIP)

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (might be obsolete with CI later on)
- [ ] New and existing unit tests pass locally with my changes (might be obsolete with CI later on)

### CODE REVIEW

File: doc/03_research/02_perception/first_implementation_plan.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  * [Lane Detection](#lane-detection)
+  * [Traffic Light Detection](#traffic-light-detection)
+  * [Traffic Sign Detection](#traffic-sign-detection)
+  * [Prediction](#prediction)
+<!-- TOC -->
+
+---
+## Overview
+
+![](../../00_assets/Umsetzungsplan Perception.jpg)
```
Comment (timdreier) [inline doc/03_research/02_perception/first_implementation_plan.md:39]: GitHub doesn't render this image. Probably the space is the problem. Best would be to remove it and use a english filename.

File: doc/03_research/02_perception/first_implementation_plan.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+## Traffic Light Detection 
+
+This module is responsible for the detection of traffic lights and the classification of their current state. \
+It combines the information about position given by the map and Obstacle Detection and Object Classification Module and 
+the information about the state given by the camera images.
+
+For state classification, the part showing the traffic light is snipped out of the camera image to get a better and 
+faster prediction due to a smaller image size and less information. To prevent the classification network to classify 
+the state always as orange on traffic lights with orange housings, the housing is masked out. 
+As classification net I would recommend the net implemented by PAF21-1.
```
Comment (timdreier) [inline doc/03_research/02_perception/first_implementation_plan.md:118]: Is there a link to the according code or even documentation to this implementation?

**Review by timdreier — CHANGES_REQUESTED** [review]
Pretty good summary overall!
Some parts are missing / have no plan yet. Should we create new issues for them, so that they won't get lost?

`b5 lint` still throws some errors which should be fixed before merging. Probably this can be done automatically by running `b5 markdown:fix`.

Time to Review ~30 min.

**Comment by timdreier** [comment]
And maybe rename `first_implementation_plan.md` to `03_first_implementation_plan.md` to keep a persistent ordering? :)

File: doc/03_research/02_perception/first_implementation_plan.md (hunk 2 above)
Comment (MarcoRiedenauer) [reply doc/03_research/02_perception/first_implementation_plan.md]: Solved in [cf3092b](https://github.com/ll7/paf22/pull/59/commits/cf3092bacaf21e954fab86c12c076c9a1044ae03)

File: doc/03_research/02_perception/first_implementation_plan.md (hunk 1 above)
Comment (MarcoRiedenauer) [reply doc/03_research/02_perception/first_implementation_plan.md]: Solved in [cf3092b](https://github.com/ll7/paf22/pull/59/commits/cf3092bacaf21e954fab86c12c076c9a1044ae03)

**Pull request merged**